### PR TITLE
Amam/add p2p cashier

### PR DESCRIPTION
**Pull request by mustofa-binary**
- add dp2p info note to platform [mobile]
- add dp2p info note to platform [web]
- add 2 new images

### CODE REVIEW

File: src/javascript/app/pages/cashier/cashier.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ const Cashier = (() => {
             $toggler.toggleClass('open');
         });
         showCashierNote();
+        showP2PNote();
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: this should be onLoad not in showcontent. we're not showing it in paymentmethods page so it doesn't need to be here.

File: src/javascript/app/pages/cashier/cashier.js [hunk 2]
```diff
@@ -43,6 +44,16 @@ const Cashier = (() => {
         });
     };
 
+    const showP2PNote = () => {
+        BinarySocket.wait('authorize').then(() => {
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: when you move this to onload we don't need to wait for authorize anymore

File: src/javascript/app/pages/cashier/cashier.js [hunk 3]
```diff
@@ -43,6 +44,16 @@ const Cashier = (() => {
...
+            $('#dp2p_info').setVisibility(
+                Client.isLoggedIn() &&           // only show to logged-in clients
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: this can also be removed since onload we check that client is logged in

File: src/templates/app/cashier/index.jsx [hunk 4]
```diff
@@ -139,6 +139,30 @@ const Cashier = () => (
                 <DepositWithdrawButton is_payment_agent />
             </div>
         </div>
+
+        <div className='gr-padding-10' />
+
+        <div className='gr-padding-10 table-body invisible gr-parent' id='dp2p_info'>
+            <h3 className='gr-padding-10'>{it.L('DP2P')}</h3>
+            <div className='gr-row'>
+                <IconWithLink img_id='virtual_money_icon' img_src={it.url_for('images/pages/cashier/dp2p_logo.svg')} />
```
Comment (negarn) [inline src/templates/app/cashier/index.jsx]: why are we repeating the img_id?

Comment (negarn) [reply src/templates/app/cashier/index.jsx]: I mean, why is it called virtual_money_icon

File: src/templates/static/platforms.jsx [hunk 5]
```diff
@@ -194,6 +194,15 @@ const Platform = () => (
...
                             />
+                            <Platforms
+                                image='dp2p-app'
+                                header={it.L('DP2P app')}
+                                description={it.L('Peer-to-peer fiat exchange')}
+                                text={it.L('The fast, easy, and convenient way to make deposits and withdrawals with zero commission fees.')}
+                                url='https://to-be-announced.apk'
+                                button_text={it.L('Download DP2P app')}
+                                download='true'
+                            />
```
Comment (negarn) [inline src/templates/static/platforms.jsx]: don't add this, please comment it. mobile app has a long way to go to be ready

File: src/templates/app/cashier/index.jsx (hunk 4 above)
Comment (mustofa-binary) [reply src/templates/app/cashier/index.jsx]: my mistake, should not need img_id for this

File: src/javascript/app/pages/cashier/cashier.js [hunk 6]
```diff
@@ -43,6 +43,11 @@ const Cashier = (() => {
         });
     };
 
+    const showP2PNote = (currency) => {
+        const is_usd_account = currency === 'USD';
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: let's name it something more general like `is_currency_allowed` in case we add more supported currencies later

File: package-lock.json [hunk 7]
```diff
@@ -2841,7 +2841,7 @@
         "get-value": "^2.0.6",
         "has-value": "^1.0.0",
         "isobject": "^3.0.1",
-        "set-value": "^2.0.1",
+        "set-value": "^2.0.0",
```
Comment (negarn) [inline package-lock.json]: why these changes?

File: src/templates/app/cashier/index.jsx [hunk 8]
```diff
@@ -139,6 +139,30 @@ const Cashier = () => (
...
+                <div className='gr-6 gr-8-m'>
+                    <span>{it.L('Make a deposit or withdrawal on your account by buying and selling fiat currencies from anywhere in the world.')}</span>
+                </div>
+                <div className='gr-2 gr-12-m'>
+                    <SeparatorLine className='gr-parent gr-hide gr-show-m gr-padding-10' invisible />
+                    <div className='gr-row gr-row-align-left gr-row-align-right-m'>
+                        <div className='gr-adapt'>
+                            <Button
+                                className='toggle button'
+                                href={'#TODO'}
```
Comment (negarn) [inline src/templates/app/cashier/index.jsx]: we can use the proper link now since the p2p implementation was merged to binary-static

File: src/templates/app/cashier/index.jsx [hunk 9]
```diff
@@ -139,6 +139,30 @@ const Cashier = () => (
                 <DepositWithdrawButton is_payment_agent />
             </div>
         </div>
+
+        <div className='gr-padding-10' />
+
+        <div className='gr-padding-10 table-body invisible gr-parent' id='dp2p_info'>
+            <h3 className='gr-padding-10'>{it.L('DP2P')}</h3>
+            <div className='gr-row'>
+                <IconWithLink img_src={it.url_for('images/pages/cashier/dp2p_logo.svg')} />
```
Comment (negarn) [inline src/templates/app/cashier/index.jsx]: can you rename the file to `dp2p-logo`? the other images in this file are using dashes not underscores

File: src/templates/static/platforms.jsx [hunk 10]
```diff
@@ -194,6 +194,16 @@ const Platform = () => (
                                 button_text={it.L('Download Tick Trade App')}
                                 download='true'
                             />
+                            {/* TODO: Uncomment this when P2P Mobile app is ready */}
+                            {/* <Platforms
+                                image='dp2p-app'
+                                header={it.L('DP2P app')}
+                                description={it.L('Peer-to-peer fiat exchange')}
+                                text={it.L('The fast, easy, and convenient way to make deposits and withdrawals with zero commission fees.')}
+                                url='https://to-be-announced.apk'
```
Comment (negarn) [inline src/templates/static/platforms.jsx]: let's leave it blank for now instead of a fake url?
```suggestion
                                url=''
```

File: src/javascript/app/pages/cashier/cashier.js [hunk 11]
```diff
@@ -43,6 +43,12 @@ const Cashier = (() => {
         });
     };
 
+    const showP2PNote = (currency) => {
+        const is_currency_allowed = currency === 'USD';
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: we should check the same logic as https://github.com/binary-com/deriv-app/pull/642/files#diff-bfde5894a635c6c5dc443bf9ed1f678fR155

just USD is not enough, client should also be either an agent or have at least one buy/sell offer available.

File: src/javascript/app/pages/cashier/cashier.js [hunk 12]
```diff
@@ -43,6 +43,19 @@ const Cashier = (() => {
         });
     };
 
+    const showP2PNote = async () => {
+        const is_agent = !(await BinarySocket.send({ p2p_agent_info: 1 })).error;
+        const has_buy = await checkP2PHasOffer('buy');
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: IMO we should check is_agent first, so in case it's true we don't need to make extra calls

File: src/javascript/app/pages/cashier/cashier.js [hunk 13]
```diff
@@ -43,6 +43,19 @@ const Cashier = (() => {
...
+        const has_sell = await checkP2PHasOffer('sell');
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: same here, we should check if has_buy first so we don't make extra calls if it's not needed

File: src/javascript/app/pages/cashier/cashier.js [hunk 14]
```diff
@@ -149,6 +162,7 @@ const Cashier = (() => {
                         State.getResponse('statement'),
                         State.getResponse('mt5_login_list')
                     );
+                    showP2PNote();
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: can we rename this to setP2PVisibility or similar? since it's not necessarily showing it in all cases

File: src/javascript/app/pages/cashier/cashier.js [hunk 15]
```diff
@@ -43,6 +43,30 @@ const Cashier = (() => {
         });
     };
 
+    const showP2PVisibility = async () => {
+        const is_agent = !(await BinarySocket.send({ p2p_agent_info: 1 })).error;
+        if (is_agent) {
+            $('#dp2p_info').setVisibility(true);
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: generally we use 0 and 1 instead of true/false.
```suggestion
            $('#dp2p_info').setVisibility(1);
```

File: src/javascript/app/pages/cashier/cashier.js [hunk 16]
```diff
@@ -43,6 +43,30 @@ const Cashier = (() => {
...
+            return;
+        }
+
+        const has_buy = await checkP2PHasOffer('buy');
+        if (has_buy) {
+            $('#dp2p_info').setVisibility(true);
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]:
```suggestion
            $('#dp2p_info').setVisibility(1);
```

File: src/javascript/app/pages/cashier/cashier.js [hunk 17]
```diff
@@ -43,6 +43,30 @@ const Cashier = (() => {
...
+            return;
+        }
+
+        const has_sell = await checkP2PHasOffer('sell');
+        if (has_sell) {
+            $('#dp2p_info').setVisibility(true);
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]:
```suggestion
            $('#dp2p_info').setVisibility(1);
```

File: src/javascript/app/pages/cashier/cashier.js [hunk 18]
```diff
@@ -149,6 +173,7 @@ const Cashier = (() => {
                         State.getResponse('statement'),
                         State.getResponse('mt5_login_list')
                     );
+                    showP2PVisibility();
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: set instead of show?

Comment (negarn) [reply src/javascript/app/pages/cashier/cashier.js]:
```suggestion
                    setP2PVisibility();
```

File: src/templates/app/cashier/index.jsx [hunk 19]
```diff
@@ -139,6 +139,30 @@ const Cashier = () => (
                 <DepositWithdrawButton is_payment_agent />
             </div>
         </div>
+
+        <div className='gr-padding-10' />
+
+        <div className='gr-padding-10 table-body invisible gr-parent' id='dp2p_info'>
+            <h3 className='gr-padding-10'>{it.L('DP2P')}</h3>
```
Comment (negarn) [inline src/templates/app/cashier/index.jsx]: are we calling it DP2P in binary? let's discuss this

Comment (mustofa-binary) [reply src/templates/app/cashier/index.jsx]: have we discuss this yet?

File: src/sass/static/pages.scss [hunk 20]
```diff
@@ -1641,4 +1641,15 @@ body #not_authenticated_financial ul.checked > li {
...
     font-size: 10px;
     position: relative;
+
+    @media screen and (max-width: 1023px) {
+        display: none;
+    }
+}
+
+#message_dp2p_mobile_view {
+    @media screen and (max-width: 1023px) {
```
Comment (negarn) [inline src/sass/static/pages.scss]: no need for these, look at the classes `gr-hide` and `gr-show-m`

File: src/javascript/app/pages/cashier/cashier.js [hunk 21]
```diff
@@ -43,6 +43,24 @@ const Cashier = (() => {
         });
     };
 
+    const setP2PVisibility = async () => {
+        const is_agent = !(await BinarySocket.send({ p2p_agent_info: 1 })).error;
```
Comment (negarn) [inline src/javascript/app/pages/cashier/cashier.js]: should we also check `is_mobile` here?

File: src/sass/static/pages.scss (hunk 20 above)
Comment (mustofa-binary) [reply src/sass/static/pages.scss]: the design broke on 1023px and below

File: src/templates/app/cashier/index.jsx (hunk 19 above)
Comment (negarn) [reply src/templates/app/cashier/index.jsx]: yes, please change it to P2P

File: src/javascript/app/pages/cashier/cashier.js (hunk 21 above)
Comment (mustofa-binary) [reply src/javascript/app/pages/cashier/cashier.js]: in this case, we won't show p2p at all on mobile?

Comment (negarn) [reply src/javascript/app/pages/cashier/cashier.js]: no, in cashier page we won't show p2p at all, and in p2p page we will show a message in case someone goes on that page by direct link

Comment (mustofa-binary) [reply src/javascript/app/pages/cashier/cashier.js]: then in that case we will need to add again when p2p mobile is ready

Comment (negarn) [reply src/javascript/app/pages/cashier/cashier.js]: yes, that's why I asked to add todo comments :)
